src/main/java/core/basesyntax/impl/StorageImpl.java

```diff
@@ -3,17 +3,50 @@
 import core.basesyntax.Storage;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private static final int MAX_ITEMS_NUMBER = 10;
+    private final K[] keys;
+    private final V[] values;
+    private int size;
+
+    public StorageImpl() {
+        this.keys = (K[]) new Object[MAX_ITEMS_NUMBER];
+        this.values = (V[]) new Object[MAX_ITEMS_NUMBER];
+    }
+
     @Override
     public void put(K key, V value) {
+        int index = -1;
+        index = findKeyIndex(key);
+        if (index > -1) {
+            keys[index] = key;
+            values[index] = value;
+        } else if (size < MAX_ITEMS_NUMBER) {
+            keys[size] = key;
+            values[size] = value;
+            size++;
+        }
     }
 
     @Override
     public V get(K key) {
+        int index = findKeyIndex(key);
+        if (index > -1) {
+            return values[index];
+        }
         return null;
     }
 
     @Override
     public int size() {
+        return size;
+    }
+
+    private int findKeyIndex(K key) {
+        for (int i = 0; i < keys.length; i++) {
+            if ((keys[i] == key && values[i] != null) || (key != null && key.equals(keys[i]))) {
```

Comment:
```suggestion
            if (keys[i] == key && values[i] != null || key != null && key.equals(keys[i])) {
```
redundant brackets. Read about logical operators' priority